### PR TITLE
71434 Record values with explicit types

### PR DESCRIPTION
**Pull request by janid1967**
In today solution, client send double and bool values as strings when recording values for fields. 
This is a refactor of this, so client send actual double and bool values to avoid string parsing.

**Breaking change for client. This PR will not / can not be merged before Mars demo**

### CODE REVIEW

**Comment by janid1967** [comment]
@kfaln This is the new code I talked about. Can You take a look at this PR and see if you can find time to take the UI part of it? UI PBI is 72371 (estimate 12h, but I guess You use less)

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/ProjectAggregate/PreservationPeriod.cs [hunk 1]
```diff
@@ -106,36 +106,47 @@ public void SetComment(string comment)
...
+            if (field.FieldType != FieldType.Number)
             {
-                return;
+                throw new Exception($"Can't record a {nameof(FieldType.Number)} value for a {field.FieldType} field");
             }
+            ValidateAndPrepareForNewRecording(field);
 
-            switch (field.FieldType)
+            // save new value ONLY if there is a value!
+            if (value.HasValue)
```
Comment (kfaln) [inline src/Equinor.Procosys.Preservation.Domain/AggregateModels/ProjectAggregate/PreservationPeriod.cs:133]: What happens when a user "blanks" out a number field? It seems this check prevents the new (blank) value from being stored. 

Any previous value is removed in `ValidateAndPrepareForNewRecording`.

Afterwards, when reading back the value in UI, I think the default `double` value 0 is returned.

Comment (janid1967) [reply src/Equinor.Procosys.Preservation.Domain/AggregateModels/ProjectAggregate/PreservationPeriod.cs:133]: Have U tested it from client? The backend logic is: when we had a value, and blank comes in, the old value should be removed and no new record should be saved.
I think I have unit test for that case, but will double-check it

Comment (janid1967) [reply src/Equinor.Procosys.Preservation.Domain/AggregateModels/ProjectAggregate/PreservationPeriod.cs:133]: I can't reproduce. I tested it against db in Azure on Tag with ID 52 (1301-E005). Using Swagger and recording for requirement with id 64 with this payload:

{
  "numberValues": [
    {
      "fieldId": 6,
      "value": 99,
      "isNA": false
    }
  ],
  "checkBoxValues": [],
  "comment": "none"
}

and afterwards

{
  "numberValues": [
    {
      "fieldId": 6,
      "value": null,
      "isNA": false
    }
  ],
  "checkBoxValues": [],
  "comment": "none"
}

... the 99-value are deleted from DB.

Comment (kfaln) [reply src/Equinor.Procosys.Preservation.Domain/AggregateModels/ProjectAggregate/PreservationPeriod.cs:133]: I will do some more testing.

File: src/Equinor.Procosys.Preservation.WebApi/Controllers/Tags/TagsController.cs [hunk 2]
```diff
@@ -273,13 +273,20 @@ public async Task<ActionResult<Query.CheckAreaTagNo.AreaTagDto>> CheckAreaTagNo(
             [FromRoute] int requirementId,
             [FromBody] RequirementValuesDto requirementValuesDto)
         {
-            var fieldValues = requirementValuesDto?
-                .FieldValues
-                .ToDictionary(
-                    keySelector => keySelector.FieldId,
-                    elementSelector => elementSelector.Value);
+            var numberValues = requirementValuesDto?
```
Comment (kfaln) [inline src/Equinor.Procosys.Preservation.WebApi/Controllers/Tags/TagsController.cs:276]: The method name `RecordCheckBoxChecked` might not be entirely accurate...

File: src/Equinor.Procosys.Preservation.Domain/AggregateModels/ProjectAggregate/PreservationPeriod.cs (hunk 1 above)
Comment (kfaln) [reply src/Equinor.Procosys.Preservation.Domain/AggregateModels/ProjectAggregate/PreservationPeriod.cs:133]: My bad. Error was in UI.

**Pull request merged**